services/leecher/Dockerfile

```diff
@@ -1,11 +1,12 @@
 # In order for this Dockerfile to work it needs to be built from the `services` directory
-FROM python:alpine3.18
+FROM python:alpine3.16
```

Comment:
I'd rather we did use
`FROM python:3.11-alpine3.18`

Reply:
It doesn't work for me on Windows and I really don't know why:

```
[+] Building 0.7s (7/21)                                                                                                            docker:default
 => [internal] load build definition from Dockerfile                                                                                          0.0s
 => => transferring dockerfile: 1.32kB                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                             0.0s
 => => transferring context: 2B                                                                                                               0.0s
 => [internal] load metadata for docker.io/library/python:3.11-alpine3.18                                                                     0.5s
 => [ 1/17] FROM docker.io/library/python:3.11-alpine3.18@sha256:537b7980bc438b55f3a27806e034ba50cb8e45338596e3b1928e6b25c1169a60             0.0s
 => [internal] load build context                                                                                                             0.0s
 => => transferring context: 22.73kB                                                                                                          0.0s
 => CACHED [ 2/17] RUN apk add --update --no-cache git curl build-base libffi-dev py3-pip                                                     0.0s
 => ERROR [ 3/17] RUN python -m ensurepip                                                                                                     0.1s
------
 > [ 3/17] RUN python -m ensurepip:
------
Dockerfile:8
--------------------
   6 |     # Install python/pip
   7 |     RUN apk add --update --no-cache git curl build-base libffi-dev py3-pip
   8 | >>> RUN python -m ensurepip
   9 |     RUN python -m pip install --no-cache --upgrade pip setuptools poetry pip-autoremove && ln -sf pip3 /usr/bin/pip
  10 |
--------------------
ERROR: failed to solve: process "/bin/sh -c python -m ensurepip" did not complete successfully: exit code: 135
```

but with just `python:alpine3.16` it is ok.

Reply:
I just built it in the Windows test machine:
```ps
PS C:\Users\ynput\Projects\ayon-shotgrid\services> docker build --network=host --no-cache -t ynput/ayon-shotgrid-processor:0.2.4 -f processor/Dockerfile .
2023/10/24 18:13:39 http2: server: error reading preface from client //./pipe/docker_engine: file has already been closed
[+] Building 52.9s (23/23) FINISHED                                                                                                                                                               docker:default
 => [internal] load .dockerignore                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                             0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                        0.0s
 => => transferring dockerfile: 1.31kB                                                                                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/python:3.11-alpine3.18                                                                                                                                   0.0s
 => [internal] load build context                                                                                                                                                                           0.1s
 => => transferring context: 1.12kB                                                                                                                                                                         0.0s
 => [ 1/18] FROM docker.io/library/python:3.11-alpine3.18                                                                                                                                                   0.1s
 => [ 2/18] RUN apk add --update --no-cache git curl build-base libffi-dev                                                                                                                                  4.0s
 => [ 3/18] RUN python -m ensurepip                                                                                                                                                                         3.1s
 => [ 4/18] RUN python -m pip install --no-cache --upgrade pip setuptools poetry pip-autoremove && ln -sf pip3 /usr/bin/pip                                                                                12.2s
 => [ 5/18] RUN mkdir /service                                                                                                                                                                              0.2s
 => [ 6/18] COPY processor/pyproject.toml /service/pyproject.toml                                                                                                                                           0.0s
 => [ 7/18] WORKDIR /service                                                                                                                                                                                0.0s
 => [ 8/18] RUN poetry config virtualenvs.create false                                                                                                                                                      0.4s
 => [ 9/18] RUN poetry install --no-interaction --no-ansi --only main                                                                                                                                      27.1s
 => [10/18] RUN apk del git curl build-base libffi-dev                                                                                                                                                      0.2s
 => [11/18] RUN python -m pip cache purge                                                                                                                                                                   0.5s
 => [12/18] RUN pip-autoremove -y setuptools poetry                                                                                                                                                         2.2s
 => [13/18] RUN python -m pip uninstall -y pip pip-autoremove                                                                                                                                               0.8s
 => [14/18] RUN addgroup -S ayonuser && adduser -SH ayonuser -G ayonuser                                                                                                                                    0.2s
 => [15/18] RUN chown ayonuser:ayonuser -R /service                                                                                                                                                         0.1s
 => [16/18] RUN chmod 777 -R /service                                                                                                                                                                       0.1s
 => [17/18] COPY processor/processor /service/processor                                                                                                                                                     0.0s
 => [18/18] COPY shotgrid_common /service                                                                                                                                                                   0.0s
 => exporting to image                                                                                                                                                                                      1.4s
 => => exporting layers                                                                                                                                                                                     1.4s
 => => writing image sha256:ce96a3606260630135f9f4c2406f147b1e3210b6749282a7ff112e105a5f9434                                                                                                                0.0s
 => => naming to docker.io/ynput/ayon-shotgrid-processor:0.2.4
 ```